eureka-client/src/main/java/com/netflix/discovery/InstanceInfoReplicator.java

```diff
@@ -68,10 +69,27 @@ public void start(int initialDelayMs) {
     }
 
     public void stop() {
-        scheduler.shutdownNow();
+        Future latestPeriodic = scheduledPeriodicRef.get();
+        if (latestPeriodic != null && !latestPeriodic.isDone()) {
+            logger.info("Canceling the latest scheduled before stop");
+            boolean result = latestPeriodic.cancel(false);
+            System.out.println("Result:" + result);
```

Comment:
If keeping this, should use logger and reformat the message to give more context.